### PR TITLE
✎ Rewrite Overview & other documentation updates

### PR DESCRIPTION
**Pull request by jmfayard**
@LouisCAD please read the new Overview page
and maybe let's discuss it by phone

https://jmfayard.github.io/refreshVersions/

### CODE REVIEW

**Comment by LouisCAD** [comment]
I'll review it this weekend. I need to release 0.9.7 properly (and the release process needs to be updated now that we got the website), and then I need to sleep and finish the office week.

**Comment by LouisCAD** [comment]
@jmfayard Can you get PRs and the non main branch to deploy on another base url so we don't impact the production site that can be updated by new releases happening on the main branch?

File: mkdocs.yml [hunk 1]
```diff
@@ -5,16 +5,29 @@ site_name: gradle refreshVersions
...
+    github: https://github.com/jmfayard/refreshVersions
+    master: https://github.com/jmfayard/refreshVersions/blob/master
+    issues: https://github.com/jmfayard/refreshVersions/issues
+    gradlePluginPortal: https://plugins.gradle.org/plugin/de.fayard.refreshVersions
+    slack: https://app.slack.com/client/T09229ZC6/CP5659EL9
   version:
-    gradle: 6.6.1
+    gradle: 6.7
     refreshVersions: '0.9.6'
     refreshVersionsDev: '0.9.6-dev-007'
```
Comment (LouisCAD) [inline mkdocs.yml:25]:
```suggestion
    refreshVersions: '0.9.7'
    refreshVersionsDev: '0.9.7-dev-001'
```

**Comment by jmfayard** [comment]
For the README, once we are happy with the website, I am inclined to do like runningcode and replace it with a big link to the documentation. Very effective remedy to the "people don't click on link" issue

image

**Comment by LouisCAD** [comment]
@jmfayard Why do you use the "auto-magically" wording instead of automatically or something more precise in the context? Using the work "magically" is a little contradictory with the "No magic!" heading in the Overview page.

If there's no deep reason, I'm considering changing it.

**Comment by LouisCAD** [comment]
Everything but the contributing docs should be alright (their content is `TK` following convention in the newspaper industry, to easily find it back).

I need to sleep now, so I'll leave it to later.

Here's the full guide I had started: https://github.com/jmfayard/refreshVersions/blob/dea553263ed5a0a2bad38e2888b9db479b687837/CONTRIBUTING.md

Feel free to start filling the contributing docs taking inspiration from the draft linked just above, and this great docs by LeakCanary: https://square.github.io/leakcanary/how_to_help/ I'll review updates to this PR starting today after I sleep enough hours.

**Comment by LouisCAD** [comment]
Here's another question: What do you plan to put in "Common errors"?

**Comment by jmfayard** [comment]
Thanks for your improvements!

Didn't know about `TK`

Will think about the contributing guide.

We need to talk about migrations.

> Common erros

Will replace it with FAQ

**Comment by jmfayard** [comment]
Looks good to me. 
The contributing guide needs work but I propose to do it in a separate pull-request

File: docs/setup.md [hunk 2]
```diff
@@ -2,7 +2,7 @@
 
 This guide will help you set up refreshVersions in a Gradle project.
 
-## Ensure Gradle is up to date
+## Update Gradle
```
Comment (LouisCAD) [inline docs/setup.md:5]: This is outside of the scope of what the commit title specifies.

Why this change?

**Comment by jmfayard** [comment]
TODO afer call

- [ ] Overview: Add an example with `Ktor` where there is `AndroidX` to have a server-side example
- [ ]  We saw this made developers to update the dependencies less and less often, effectively leading to legacy accumulation (a form of technical debt), or even projects abandon.
- [ ]  Slow badge: use  `{{version.refreshVersions}}` instead of parsing maven matadata
- [ ]  Use triple backsticks for things like `./gradlew wrapper --gradle-version 6.7` to have copy/paste

**Review by LouisCAD — APPROVED** [review]
Alright, I think the website is really good now!

We'll address the contributing guide later.

Merging this.
